src/storage/query/GetPropProcessor.cpp

```diff
@@ -186,7 +186,8 @@ nebula::cpp2::ErrorCode GetPropProcessor::buildTagContext(const cpp2::GetPropReq
     } else {
         // not use const reference because we need to modify it when all property need to return
         auto returnProps = std::move(*req.vertex_props_ref());
-        ret = handleVertexProps(returnProps);
+        auto lackTags = lackTag(returnProps);  // Check the vid
```

Comment:
Could consider add a flag in `GetPropRequest`, for now, the `lockTags` will force **every request** to check if each tag data exists. Do we really to do that? At least, at a todo here.

Please check graph ci as well.

Reply:
The graph test is ok in my local running. And in fact, we don't when to set the flag which determine whether scan all tags.

Reply:
>  And in fact, we don't know when to set the flag which determine whether scan all tags.

Why? As your example, `match (a:player)--(b)--(c) return c.age`. It is `c.age`, not `player.age. 

Also, `fetch prop on *` is different from `fetch prop on player`. Why we can't tell the difference.

Reply:
There is no property of vertex, we just have property of tag. So if only scan the `player` tag, some exist vertex will be ignored.

Reply:
Why graph can't set the flag which determine whether scan all tags. **The query can tell the difference between `scan all tags` and `only one tag`**

Reply:
We don't know when to set the flag to be true. Because in graph, we don't known which tag is attached to vertex.
For example:
```
fetch prop on player 'xxx'
```
Some vertex has player tag, some don't. And graph don't know it.

Reply:
`fetch prop on player 'xxx`` -> query has specify only tag `player`
`fetch prop on * 'xxx'` -> query has specify all tags

Why we don't know

Reply:
If we only scan the player tag, if it's not attached to the vertex, this don't return one row.

Reply:
We could modify it to return the row. But this is not the reason to read all tags all the time.

Reply:
Check your PR #160, in that PR you don't return the row, and it involves another read. I reverted in #331 after discuss. Why we back and forth again and again? That is why we need to set a interface which can be specified **explicitly**, no more default behavior.

Reply:
It's required by https://github.com/vesoft-inc/nebula-graph/pull/1137, in this match we only collect the property used, so if the request tag is not attached to the vertex, the vertex won't return one row. It's mistake in match.

Reply:
And you could check this pr https://github.com/vesoft-inc/nebula-graph/pull/1150, in this case, shall we return the row?

Reply:
vesoft-inc/nebula-graph#1137, If match need to return a row even if there is no tag attached to it, I'm OK to return a row with empty. 

vesoft-inc/nebula-graph#1150 is another issue, return or not both LGTM.  The case you add seems to work by coincidence, try `FETCH PROP ON person "3" YIELD person.id`. I believe no row is returned. 

That is why I insist query engine need to tell the behavior to storage.

Reply:
This won't be included in next version. Ok, I will add a flag and set true only in match.

Reply:
👍

Reply:
Modified.